projects/storefrontlib/src/cms-components/myaccount/my-coupons/my-coupons.component.service.ts

```diff
@@ -32,7 +32,11 @@ export class MyCouponsComponentService {
         cxRoute: 'search',
         params: { query: this.buildSearchParam(coupon) },
       },
-      { couponcode: coupon.couponId }
+      {
+        state: {
+          couponcode: coupon.couponId,
+        },
+      }
```

Comment:
Previously, we passed query params in the 2nd argument `query`: `{ couponcode: coupon.couponId }`
Now, when the obsolete 2nd argument is removed, we should pass query params in the property `queryParams` of the remaining `NavigationExtras` object argument (see [docs](https://angular.io/api/router/UrlCreationOptions#queryParams)):

```suggestion
      {
        queryParams: {
          couponcode: coupon.couponId,
        },
      }
```